playwright-tests/tests/monthlyContribution.test.ts

```diff
@@ -0,0 +1,62 @@
+import { expect, test } from "@playwright/test";
+import { setTestCookies } from "./utils/cookies";
+import { email, firstName, lastName } from "./utils/users";
+
+test.beforeEach(async ({ page, context, baseURL }) => {
+  const baseUrlWithFallback = baseURL ?? "https://support.theguardian.com";
+  const pageUrl = `${baseUrlWithFallback}/uk/contribute`;
+
+  const domain = new URL(pageUrl).hostname;
+  await setTestCookies(context, firstName, domain);
+  await page.goto(pageUrl, {waitUntil: "networkidle"});
+});
+
+test.describe("Sign up for a Recurring Contribution (New Contributions Flow)", () => {
+  test("Monthly contribution sign-up with Stripe - GBP", async ({ page }) => {
+    const monthlyTab = "#MONTHLY";
```

Comment:
Happy to open the discussion wider as it's more abstract, and could be part of a second PR where we could have this discussion.

Playwright [have a list of preferred locators](https://playwright.dev/docs/locators) - from experience and the literature, it's mainly to do so that you test as closely to what the user sees as possible, allowing us to think about the accessibility as we go and hopefully reduce releasing bugs that the user sees, but we don't as it's obfuscated through invisible IDs.

Testing library cite a great quote:
> [The more your tests resemble the way your software is used, the more confidence they can give you.](https://testing-library.com/docs/guiding-principles)

As per Playwrights docs:
> To make tests resilient, we recommend prioritizing user-facing attributes and explicit contracts such as [page.getByRole()](https://playwright.dev/docs/api/class-page#page-get-by-role).

– [Playwright docs](https://playwright.dev/docs/locators)

What do we think about following a similar pattern?